src/rust-crypto/CrossSigningIdentity.ts

```diff
@@ -87,12 +87,18 @@ export class CrossSigningIdentity {
                     "bootstrapCrossSigning: Cross-signing private keys not found locally, but they are available " +
                         "in secret storage, reading storage and caching locally",
                 );
-                await this.olmMachine.importCrossSigningKeys(
+                const status: CrossSigningStatus = await this.olmMachine.importCrossSigningKeys(
                     masterKeyFromSecretStorage,
                     selfSigningKeyFromSecretStorage,
                     userSigningKeyFromSecretStorage,
                 );
 
+                // Check that `importCrossSigningKeys` worked correctly (for example, it will fail silently if the
+                // public keys are not available).
+                if (!status.hasMaster || !status.hasSelfSigning || !status.hasSelfSigning) {
```

Comment:
This seems to check the same thing twice

Reply:
gah, good spot

```suggestion
                if (!status.hasMaster || !status.hasSelfSigning || !status.hasUserSigning) {
```